### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by lincoln1883**
For this project milestone, I completed the following tasks: 
- Create a class Nameable
  - [ ] Implement a method called correct_name that will raise a NotImplementedError
- Turn your Person class to Nameable
  - [ ] Person class inherits from Nameable.
  - [ ] Person class has a method correct_name implemented. It should simply return the name attribute.
- Prepare base Decorator.
  - [ ] Make sure that it inherits from Nameable.
  - [ ] In the constructor assign a nameable object from params to an instance variable.
  - [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable
- Prepare CapitalizeDecorator and TrimmerDecorator
  - [ ] Create a class that inherits from the base Decorator class.
  - [ ] Implement a method correct_name that capitalizes the output of @nameable.correct_name
  - [ ] Implement a method correct_name that ensures that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
```
Comment (zieeco) [inline base_decorator.rb:3]: - Hey, let's add the setter/getter method for the nameable so it could be read `attr_accessor :nameable`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @lincoln1883,




Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter error.✔️
- Nice decorating the classes. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (lincoln1883) [reply base_decorator.rb:3]: Done.

**Review by Munsa1 — APPROVED** [review]
Hi @lincoln1883, 
Approved 1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Change By Previous Reviewer :heavy_check_mark: 


## Optional suggestions

- [ ] Kindly consider removing the `Live Demo` section from your readme file as there is no live link available to share, for a more professional readme file.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**